File: SGEM/Models/AsignacionEquipo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SGEM.Models
{
    public class AsignacionEquipo : Base
    {
        public DateTime FechaCreacion { get; set; }
        public Usuario UsuarioCreacion { get; set; }
        public DateTime FechaAprobacion { get; set; }
        public Usuario UsuarioAprobacion { get; set; }
        public Almacen Almacen { get; set; }
        public Persona Empleado { get; set; }
        public IList<AsignacionEquipoItem> Items { get; set; }

        public AsignacionEquipo()
        {
            Items = new List<AsignacionEquipoItem>();
        }
    }
}

File: SGEM/ClientApp/src/app/asignaciones-equipo/asignacion-equipo.ts
import { IAlmacen } from "../almacenes/almacen";
import { IUsuario } from "../usuarios/usuario";
import { IPersona } from "../personas/persona";
import { IEquipo } from "../equipos/equipo";

export interface IAsignacionEquipo {
  id?: number;
  fechaCreacion?: Date;
  usuarioCreacion?: IUsuario;
  fechaAprobacion?: Date;
  usuarioAprobacion?: IUsuario;
  comentario?: string;
  almacen?: IAlmacen;
  empleado?: IPersona;
  items?: IAsignacionEquipoItem[];
}

export interface IAsignacionEquipoItem {
  id?: number;
  idAsignacionEquipo?: number;
  equipo?: IEquipo;
  cantidad?: number;
  costo?: number;
}
